build.gradle

```diff
@@ -269,11 +269,29 @@ subprojects {
     tasks.register('allDependencyInsight', DependencyInsightReportTask)
     tasks.register('allDependencies', DependencyReportTask)
 
-    tasks.register('listReleasableSoftware'){
+    tasks.register('releasable'){
         description = "Prints a list of all modules which will be released externally"
         group = "Release"
-        if(project.hasProperty('releasable')){
-            logger.quiet("$project.name")
+        if(project.hasProperty('releasable') && project.releasable){
+            project.publishing.publications.each { publication ->
+                logger.quiet("\n${publication.groupId}:${publication.artifactId}:${publication.version} [${project.path}]")
+                publication.artifacts.each { artifact ->
+                    logger.quiet(" * ${artifact.file.name}")
+                }
+            }
+        }
+    }
+
+    tasks.register('unReleasable'){
+        description = "Prints a list of all modules which will not be released externally"
+        group = "Release"
+        if(!project.hasProperty('releasable') || !project.releasable){
+            project.publishing.publications.each { publication ->
+                logger.quiet("\n${publication.groupId}:${publication.artifactId}:${publication.version} [${project.path}]")
+                publication.artifacts.each { artifact ->
```

Comment:
I'm hoping you find a way to reuse code instead of copy/paste

Reply:
Good point, Logic updated for more reusability and support method added.  medium term these tasks can be extracted further into a plugin but again that requires re-work of our corda-internal-gradle plugins repo which id rather not spend excess time on pre DP2 with other items still outstanding  - Tech Depth item to fully externalize these so can be leveraged by any project using R3 publish plugin https://r3-cev.atlassian.net/browse/CORE-6451